code/modules/goonhub/api/routes/maps/MapSwitches.dm

```diff
@@ -0,0 +1,8 @@
+
+/// POST /map-switch
+/// Trigger a map switch for a given server
+/datum/apiRoute/mapswitch
+	method = RUSTG_HTTP_METHOD_POST
+	path = "/map-switch"
+	body = /datum/apiBody/mapswitch
+	correct_response = list(/datum/apiModel/Tracked/MapSwitch, "status")
```

Comment:
probably best to just edit `/datum/apiModel/Tracked/MapSwitch` to be (or similar):
```cs
/datum/apiModel/Tracked/MapSwitch
	var/datum/apiModel/Tracked/MapSwitchInternal/map_switch = null
	var/status = "string"

/datum/apiModel/Tracked/MapSwitchInternal
```